pkg/operator/v1helpers/helpers.go

```diff
@@ -550,3 +554,13 @@ func IsConditionPresentAndEqual(conditions []metav1.Condition, conditionType str
 	}
 	return false
 }
+
+// IsUpdatingTooLong determines if updating operands takes too long.
+// it returns true if the progressing condition has been set to True for at least 15 minutes
+func IsUpdatingTooLong(operatorStatus *operatorv1.OperatorStatus, progressingConditionType string) (bool, error) {
+	progressing := FindOperatorCondition(operatorStatus.Conditions, progressingConditionType)
+	if progressing == nil {
+		return false, fmt.Errorf("operator status does not have %s condition", progressingConditionType)
+	}
```

Comment:
I would prefer to stay safe and return `false, nil` here, as the previous code did. Have you checked all callers? Who knows what it breaks.

Reply:
Yeah, the only caller was workload operator, however adding the error made many unit tests fail because of what you described - that the condition is not there on first sync. I have not thought of that initially and wanted to make the function more robust, I guess it was written intentionally not to check nil and return errors exactly because of this. Thank you for the hints.